### PR TITLE
[Merged by Bors] - support dns addresses

### LINKED ISSUE #3156: p2p: support dns entrypoints
Libp2p accepts DNS entrypoint for the node. Example:

```
/dns4/bootstrap-0.mainnet.filops.net/tcp/1347/p2p/12D3KooWCVe8MmsEMes2FzgTpt9fXtmCY7wrq91GRiaC8PHSCCBj
```

In the existing deployment scheme we create several nodes and then connect every other node to them. It makes deployment more complex, and more importantly, relies on IP addresses being the same all the time.

Instead, in the new deployment scheme, we will generate entrypoints in advance and share them for every node in the cluster.

### PR DESCRIPTION
**Pull request by abergasov**
## Motivation
Now for bootnodes can be used links with dns addresses.
DNS resolve address on connect
Closes #3156

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

**Review by dshulyak — COMMENTED** [review]
is it possible to add some meaningful (unit) test? for example:

1. add dns addr to the address book manually, spawn protocol server
2. query addresses from protocol server
3. check that dns address was added to the local address book

File: p2p/peerexchange/addrbook.go [hunk 1]
```diff
@@ -112,7 +113,8 @@ type addrBook struct {
 // updateAddress is a helper function to either update an address already known
 // to the address manager, or to add the address if not already known.
 func (a *addrBook) updateAddress(addr, src *addrInfo) {
-	if !IsRoutable(addr.IP) && IsRoutable(src.IP) {
+	routableAddr := IsRoutable(addr.IP) || strings.HasPrefix(addr.RawAddr, "/dns")
```
Comment (countvonzero) [inline p2p/peerexchange/addrbook.go]: how about making a func IsDNSAddr() or something like that?

**Review by dshulyak — APPROVED** [review]
linter failed

**Review by countvonzero — APPROVED** [review]

**Comment by dshulyak** [comment]
bors merge

**Pull request closed without merging**